document/core/valid/modules.rst

```diff
@@ -161,7 +161,10 @@ Element segments :math:`\elem` are classified by :ref:`segment types <syntax-seg
 * The expression :math:`\expr` must be :ref:`constant <valid-constant>`.
 
 * For each :math:`y_i` in :math:`y^\ast`,
-  the function :math:`C.\CFUNCS[y]` must be defined in the context.
+
+  * :math:`y_i` must be :math:`\EREFFUNC~z`.
```

Comment:
Can you factor this out into a separate rule for elemexpr that is shared between active/passive segment types?

I don't think you need to check here that it is a ref.func for an active segment, since that's already enforced by the binary/text format. At least in the reference interpreter we don't make that distinction during validation.